server/sonar-process-monitor/src/main/java/org/sonar/process/monitor/Monitor.java

```diff
@@ -451,7 +451,7 @@ private static void awaitTermination(Thread... threads) {
   }
 
   private static void awaitTermination(@Nullable Thread t) {
-    if (t == null) {
+    if (t == null || Thread.currentThread() == t) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1698)